src/Types.ts

```diff
@@ -253,4 +255,11 @@ export interface DecorationColorSetting {
   dark: string;
   light: string;
   highContrast: string;
-}
\ No newline at end of file
+}
+
+export type NodeFinder = (node: SyntaxNode) => SyntaxNode | null;
```

Comment:
I might move the `NodeMatcher` above down to right above this one; it got separated in the rebase

I also might add a docstring for this type and the next one as they're important abstractions